dags/gmail_data_import_pipeline.py

```diff
@@ -139,27 +147,74 @@ def gmail_link_message_thread_ids_etl(**kwargs):
 
         LOGGER.info('Created file: %s', filename)
 
-        generated_schema = generate_schema_from_file(filename)
-        LOGGER.info('generated_schema: %s', generated_schema)
-
-        create_table_if_not_exist(
-            project_name=data_config.project_name,
-            dataset_name=data_config.dataset,
+        create_or_extend_table_schema(
+            gcp_project=project_name,
+            dataset_name=dataset_name,
             table_name=table_name,
-            json_schema=generated_schema
+            full_file_location=filename,
+            quoted_values_are_strings=True
         )
-        LOGGER.info('Created table: %s', table_name)
 
         load_file_into_bq(
             filename=filename,
-            dataset_name=data_config.dataset,
+            dataset_name=dataset_name,
             table_name=table_name,
-            project_name=data_config.project_name,
+            project_name=project_name,
             auto_detect_schema=True
         )
         LOGGER.info('Loaded table: %s', table_name)
 
 
+def gmail_thread_details_etl(**kwargs):
+    data_config = data_config_from_xcom(kwargs)
+    user_id = get_gmail_user_id()
+    table_name = data_config.table_name_thread_details
+    dataset_name = data_config.dataset
+    project_name = data_config.project_name
+
+    df_thread_id_list = get_distinct_values_from_bq(
+        project_name=data_config.project_name,
+        dataset=dataset_name,
+        column_name=data_config.column_name_list_of_thread_ids,
+        table_name=data_config.table_name_list_of_thread_ids
+        )
+
+    df_thread_details = pd.concat(
+        [get_one_thread(
+            get_gmail_service(),
+            user_id,
+            id)
+            for id in df_thread_id_list[0]]
+        )
```

Comment:
minor: indentation (would move the square bracket here and the round bracket to align)